### DISCUSSION TITLE
Issue with Nginx not being able to start it

### DISCUSSION THREAD

**Discussion by facyber** [Security-Onion-Solutions/securityonion#2897] [Unsupported Versions] [answered]
Hello everyone,

I've been trying to setup Heavy Distributed SO and I downloaded latest ISO version 2.3.21 and after fresh install I noticed that there is an issue with Nginx. so-status returns an error (everything else seems good) and nginx/error.log gives me the following error message: `could not build server_names_hash, you should increase server_names_hash_bucket_size: 32`

Weird thing is that FQDN is not 32 nor more than 32 characters.

I've googled and found that possible solution is to raise value of that variable in nginx.conf file, but when I do that and restart nginx, that value is removed. What else I can try? I couldn't find anybody else had similar issue here. I haven't touched any other settings, just to know. 

In case it is related somehow to the resources, VM have 8 CPU, 16GB RAM and 200GB SSD.

Thanks

**Comment by BustedSec** [comment]
Perhaps see if the advice in this nginx thread helps https://gist.github.com/muhammadghazali/6c2b8c80d5528e3118613746e0041263

**Reply by facyber** [reply]
Hi BustedSec,

I've seen similar things and tested but not working.

What I did is:

1. Created  `/etc/nginx/conf.d/ `folders
2. Created `<something>.conf `in it
3. Added `server_names_hash_bucket_size 64; `to the file (tested also with 32 and with `http { server_names_hash_bucket_size 64; }`)
4. `so-nginx-restart`
5. Issue still present.

**Comment by weslambert** [comment, marked as answer]
I'm not sure why you would be having this issue, but I imagine your changes got overwritten or are not respected, as the nginx config is read from `/opt/so/saltstack/default/salt/nginx/etc/nginx.conf` (distributed from the manager node). 
 You could try something like `cp /opt/so/saltstack/default/salt/nginx/etc/nginx.conf /opt/so/saltstack/local/salt/nginx/etc/`, then modifying the value you mentioned, then on the heavy node perform the `so-nginx-restart`.

**Reply by facyber** [reply]
Hi @weslambert 

Thanks for the help. It seems your solution worked for me on the first try. Of course will test it further just to see will there be any implications.

Do you think in future updates can these settings be overwritten somehow? Or that shouldn't be the case?

Thanks!
